### PR TITLE
idb: update AddBlock method signature

### PR DESCRIPTION
**Pull request by shiqizng**
2nd PR for issue [1416](https://github.com/algorand/indexer/issues/1416) and a followup to #1461.
This PR updates  AddBlock interface to `func AddBlock(block *types.ValidatedBlock) error`

### CODE REVIEW

File: idb/postgres/postgres_integration_test.go [hunk 1]
```diff
@@ -2169,9 +2135,13 @@ func TestDeleteTransactions(t *testing.T) {
 	db, shutdownFunc := setupIdb(t, test.MakeGenesisV2())
 	defer shutdownFunc()
 
-	txns := []transactions.SignedTxn{}
+	txns := []sdk.SignedTxn{}
 
-	genBlock := ledgercore.MakeValidatedBlock(test.MakeGenesisBlock(), ledgercore.StateDelta{})
+	//genBlock := ledgercore.MakeValidatedBlock(test.MakeGenesisBlock(), ledgercore.StateDelta{})
```
Comment (winder) [inline idb/postgres/postgres_integration_test.go:2140]:
```suggestion
```

**Review by winder — APPROVED** [review]
Looks good, just one minor nit

File: idb/mocks/IndexerDb.go [hunk 2]
```diff
@@ -6,15 +6,13 @@ import (
...
-	ledgercore "github.com/algorand/go-algorand/ledger/ledgercore"
-
 	mock "github.com/stretchr/testify/mock"
 
 	testing "testing"
 
-	types "github.com/algorand/go-algorand-sdk/v2/types"
+	types "github.com/algorand/indexer/types"
+
+	v2types "github.com/algorand/go-algorand-sdk/v2/types"
```
Comment (Eric-Warehime) [inline idb/mocks/IndexerDb.go:15]: nit: I guess I would personally have made this `sdk "github.com/algorand/go-algorand-sdk/v2/types"`

**Review by Eric-Warehime — APPROVED** [review]

File: idb/mocks/IndexerDb.go (hunk 2 above)
Comment (shiqizng) [reply idb/mocks/IndexerDb.go:15]: unfortunately, this is auto-generated by mockery.

**Pull request merged**